Physical_Engine/Query/TopCentreline.cs

```diff
@@ -0,0 +1,69 @@
+﻿using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+using BH.Engine.Spatial;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+
+        [Description("Returns the top centreline of an IFramingElement")]
+        [Input("BHoM.Physical.Elements.IFramingElement", "The IFramingEment to query the top centreline of")]
+        [Output("ICurve", "The top centreline of the IFramingElement")]
+
+        public static ICurve TopCentreline(this BH.oM.Physical.Elements.IFramingElement element)
+        {
+            ICurve location = element.Location;
+
+            Vector normal;
+            
+            if(location.IIsLinear())
+            { 
+                normal = BH.Engine.Physical.Query.Normal(element);
+            }
+            else
+            {
+                Engine.Reflection.Compute.RecordError("IFramingElement must have linear location line");
+                normal = null;
+            }
```

Comment:
Would either change to retuning null in the else statement (as above) or change to something like

```suggestion
            normal = BH.Engine.Physical.Query.Normal(element);
            
            if(normal == null)
                return null;
```

This would make use of the checks and warnings in the normal method, and return null if it failed.

Not saying this have to be changed to this though, what you have with suggested changes to the other file works fine.